charts/redpanda/templates/statefulset.yaml

```diff
@@ -336,17 +338,20 @@ spec:
       {{- else }}
           emptyDir: {}
       {{- end }}
-      {{- if and (include "is-licensed" . | fromJson).bool .Values.storage.tieredConfig.cloud_storage_enabled }}
+      {{- if and (include "is-licensed" . | fromJson).bool (include "storage-tiered-config" .|fromJson).cloud_storage_enabled }}
+        {{- $tieredType := include "storage-tiered-mountType" . }}
+        {{- if ne $tieredType "none" }}
         - name: tiered-storage-dir
-        {{- if .Values.storage.tieredStoragePersistentVolume.enabled }}
+          {{- if eq $tieredType "persistentVolume" }}
           persistentVolumeClaim:
             claimName: tiered-storage-dir
-        {{- else if .Values.storage.tieredStorageHostPath }}
+          {{- else if eq $tieredType "hostPath" }}
           hostPath:
-            path: {{ .Values.storage.tieredStorageHostPath | quote }}
```

Comment:
This change does not guarantee that host path is not empty. Previous implementation didn't pass empty string to the host path.

Reply:
True. I'll add some validation to the `storage-tiered-mountType` template.